shared/mtemplates/templates.go

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package mtemplates
+
+import (
+	"bytes"
+	"html/template"
+	"io"
+	"path/filepath"
+	"sync/atomic"
+
+	"github.com/fsnotify/fsnotify"
+
+	"github.com/mattermost/mattermost-server/v5/mlog"
```

Comment:
I haven't done this change, because I'm not 100% sure about the best way of doing it. I'll reach you out in https://community.mattermost.com to talk about it.

Reply:
Now is ready